File: src/main/java/org/mikechabot/service/ExampleService.java
package org.mikechabot.service;

import org.mikechabot.model.Example;

import java.util.List;

/**
 * Created by mikechabot on 10/31/15.
 */
public interface ExampleService {


    List<Example> getExampleData();
    List<Example> getByName(String name);
    void createExampleData(String name);
    void clearCollection();

}

File: src/main/webapp/js/service/example_service.js
app.service('ExampleService', ['DataAccessService', function (DataAccessService) {

    var _baseServiceUrl = '/example';

    return {
        getExampleData: function() {
            return DataAccessService.get(_baseServiceUrl);
        },
        createExampleData: function(name) {
            var deferred = $.Deferred();
            if (!name) {
                deferred.reject('Name cannot be null');
            } else {
                DataAccessService.post(_baseServiceUrl + '/create', name)
                    .done(function() {
                        deferred.resolve();
                    })
                    .fail(function() {
                        deferred.reject();
                    })
            }
            return deferred.promise();
        },
        clearCollection: function() {
            return DataAccessService.post(_baseServiceUrl + '/delete/all');
        },
        findByName: function(name) {
            return DataAccessService.get(_baseServiceUrl + '/find/' + name);
        }
    };
}]);